### PR TITLE
Reorganizing and updating installation section

### PR DESCRIPTION
**Pull request by bmmpxf**
Updated the install instructions and tried to clean/simplify/streamline/unify them in the process.

### CODE REVIEW

File: doc/en/user/source/installation/win_binary.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#. Download the archive and unpack to the directory where you would like the program to be located.
+
+   .. note:: A suggested location would be :file:`C:\\Program Files\\GeoServer`.
+
+Setting environment variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You will need to set the ``JAVA_HOME`` environment variable if it is not already set. This is the path to your JRE such that :file:`%JAVA_HOME%\\bin\\java.exe` exists.
+
+.. note:: Java 7 is required, as of GeoServer 2.6.x Java 5 is no longer supported. For more information about Java and GeoServer, please see the section on :ref:`production_java`.
```
Comment (jodygarnett) [inline doc/en/user/source/installation/win_binary.rst]: This text is incomplete, like as of 2.7 java 6 is no longer supported. The section on production_java is not really the best spot for installation requirements like this ...

File: doc/en/user/source/installation/windows/binary.rst [hunk 2]
```diff
@@ -1,64 +0,0 @@
-.. _installation_windows_bin:
-
-Windows Binary
-==============
-
-.. note:: This section is for the OS-independent binary.  Please see the section on the :ref:`installation_windows_installer` for the wizard-based installer for Windows.
-
-The most common way to install GeoServer is using the OS-independent binary.  This version is a GeoServer web application (webapp) bundled inside `Jetty <http://www.mortbay.org/jetty/>`_, a lightweight servlet container system.  It has the advantages of working very similarly across all operating systems plus being very simple to set up.
```
Comment (jodygarnett) [inline doc/en/user/source/installation/windows/binary.rst]: I think metrics show the most common way is the windows installer?

**Comment by jodygarnett** [comment]
I am interested in capturing the system requirements clearly as we have users tripping up over the version of Java to use.

File: doc/en/user/source/installation/win_binary.rst (hunk 1 above)
Comment (bmmpxf) [inline doc/en/user/source/installation/win_binary.rst]: Good point, it's redundant with what's already mentioned on the page.

**Comment by bmmpxf** [comment]
Java 7 requirements (and Java 8 mentioned as a no-no) should exist on every page now.

**Pull request merged**